### ISSUE TITLE
Display actual parameters for SQL statement in Sent Message for Database Writer

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#2924]
Create a channel with a DB Writer, with an INSERT statement that takes some parameters.  Deploy and process a message through the channel.  Go to the message view and look at the Sent data for the destination.  In the [PARAMETERS] section, it doesn't show the actual listing of parameters.  Rather, it's just the basic toString() dump for a generic Java Object.  This is obviously because whatever object is being used to store the parameters (ArrayList?) doesn't implement toString().  Here's an example:

------------------------------------------------
URL: jdbc:postgresql://localhost:5432/test
USERNAME: test

[QUERY]
INSERT INTO patients (patientid, lastname, firstname, dateofbirth)
VALUES (?, ?, ?, ?)

[PARAMETERS]
[Ljava.lang.Object;@635244c2
------------------------------------------------

It would be incredibly useful to see the actual data being passed for the parameter list to the statement.  For example:

------------------------------------------------
[PARAMETERS]
[12345, FRY, PHILIP, 19730901]
------------------------------------------------

Imported Issue. Original Details:
Jira Issue Key: MIRTH-3003
Reporter: ericb
Created: 2013-10-15T10:41:42.000-0700

**Comment by rbeckman-nextgen** [comment]
We just need to change toFormattedString so that it shows the object array nicely rather than just passing it into builder.append().

Imported Comment. Original Details:
Author: narupley
Created: 2013-10-15T10:49:13.000-0700

**Comment by rbeckman-nextgen** [comment]
Revision 6973: Fixed DatabaseDispatcherProperties.toFormattedString() so that it shows each parameter nicely rather than the string representation of an Object array.

Instead of showing the parameters all on one line, we decided to break up each parameter, like so:

`[PARAMETER 1]
12345

[PARAMETER 2]
FRY

[PARAMETER 3]
PHILIP

[PARAMETER 4]
19730901`

This was done because the parameters could contain commas already, or even newlines, etc. (it's not uncommon for a single parameter to be an entire HL7 message for example).

Imported Comment. Original Details:
Author: narupley
Created: 2013-10-15T11:03:11.000-0700
